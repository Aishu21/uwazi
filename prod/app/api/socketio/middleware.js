"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports.default = void 0;var _cookie = _interopRequireDefault(require("cookie"));
var _socket = _interopRequireDefault(require("socket.io"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}var _default =

(server, app) => {
  const io = (0, _socket.default)(server);
  app.use((req, res, next) => {
    req.io = io;
    next();
  });

  app.use((req, res, next) => {
    req.getCurrentSessionSockets = () => {
      const sessionSockets = {
        sockets: [],
        emit(...args) {
          this.sockets.forEach(socket => {
            socket.emit(...args);
          });
        } };


      Object.keys(req.io.sockets.connected).reduce((sockets, socketId) => {
        const socket = req.io.sockets.connected[socketId];
        if (typeof socket.request.headers.cookie !== 'string') {
          return sockets;
        }

        const socketCookie = _cookie.default.parse(socket.request.headers.cookie);
        let sessionId;

        if (socketCookie['connect.sid']) {
          [, sessionId] = socketCookie['connect.sid'].split('.')[0].split(':');
        }

        if (sessionId === req.session.id) {
          sockets.sockets.push(socket);
        }

        return sockets;
      }, sessionSockets);

      return sessionSockets;
    };

    next();
  });
};exports.default = _default;