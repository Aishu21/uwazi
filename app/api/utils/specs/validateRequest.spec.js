import Joi from 'joi';

import middleware from '../validateRequest.js';

describe('validateRequest', () => {
  let next;
  let res;
  const req = {};

  beforeEach(() => {
    next = jasmine.createSpy('next');
    res = {
      json: jasmine.createSpy('json'),
      status: jasmine.createSpy('status'),
      error: jasmine.createSpy('error'),
    };
  });

  it('should call next when body or query is valid', () => {
    const schema = Joi.object().keys({
      prop1: Joi.string(),
      prop2: Joi.number(),
    });

    req.body = { prop1: 'string', prop2: 25 };
    middleware(schema)(req, res, next);
    expect(next).toHaveBeenCalled();

    req.query = { prop1: 'string', prop2: 25 };
    middleware(schema, 'query')(req, res, next);
    expect(next).toHaveBeenCalled();
  });

  it('should call next with the error', () => {
    const schema = Joi.object().keys({
      prop1: Joi.string(),
      prop2: Joi.number(),
    });

    req.body = { prop1: 55, prop2: 'must be number' };
    middleware(schema)(req, res, next);
    expect(next.calls.argsFor(0)).toMatchSnapshot();
  });
});
