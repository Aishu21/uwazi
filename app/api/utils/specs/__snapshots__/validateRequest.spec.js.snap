// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validateRequest should respond with an error when schema does not validate 1`] = `
Array [
  Object {
    "code": 400,
    "message": "ValidationError: child \\"prop1\\" fails because [\\"prop1\\" must be a string]",
  },
]
`;
